airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -700,6 +703,12 @@ private void deleteConnectionBeforeTerminatingTheWorkflow() {
    * Set a job as cancel and continue to the next job if and continue as a reset if needed
    */
   private void reportCancelledAndContinueWith(final boolean isReset, final ConnectionUpdaterInput connectionUpdaterInput) {
+    reportCancelled(isReset);
+    resetNewConnectionInput(connectionUpdaterInput);
+    prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
```

Comment:
Do any of these methods throw exceptions, particularly `RuntimeException`, as I know that we like to use that as a way to signal failures?  If so, should this be in a try/finally block to allow one or two of these methods to still fire if the first one or two raises an exception?

Reply:
Most of the method are getter/setter so very unlikely to throw a RuntimeException. The method that can throw are run using `runMandatoryActivity` which catch any exception and set the job in quarantine.